src/ApiInterface/include/ApiInterface/Orbit.h

```diff
@@ -369,16 +369,20 @@ inline bool orbit_api_active() {
 }  // extern "C"
 
 // Internal macros.
-#define ORBIT_CONCAT_IND(x, y) (x##y)
+#define ORBIT_CONCAT_IND(x, y) x##y
 #define ORBIT_CONCAT(x, y) ORBIT_CONCAT_IND(x, y)
 #define ORBIT_UNIQUE(x) ORBIT_CONCAT(x, __COUNTER__)
 #define ORBIT_VAR ORBIT_UNIQUE(ORB)
 
+#define ORBIT_SCOPE_WITH_COLOR_AND_GROUP_ID_INTERNAL(name, col, group_id, pc_name) \
+  uint64_t pc_name;                                                                \
+  asm("lea (%%rip), %0" : "=r"(pc_name) : :);                                      \
```

Comment:
Maybe add a comment for the asm statement.

Reply:
> Previously the intend was to get the return address after the ctor of Scope was executed

The return address was retrieved *inside* the constructor.

> Now we get the address of either the call to Scope::Scope or the first instruction of Scope::Scope (in case of inlining).

I think you are correct, but honestly I didn't think about the exact instruction too much as in the end it's all code that is hidden behind the macro.

> Maybe add a comment for the asm statement.

I hadn't because this is a public header and I didn't want to add internal documentation there, but I can do that, I guess it doesn't hurt.

Reply:
I added a comment.